lib/srv/usermgmt_linux.go

```diff
@@ -237,18 +238,55 @@ func readDefaultSkel() (string, error) {
 	return skel, trace.Wrap(err)
 }
 
+func recursiveChown(dir string, uid, gid int) error {
+	// First, walk the directory to gather a list of files and directories to update before we open up to modifications
+	var pathsToUpdate []string
+	err := filepath.WalkDir(dir, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		pathsToUpdate = append(pathsToUpdate, path)
+		return nil
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	pathsToUpdate = append(pathsToUpdate, dir)
+	// filepath.WalkDir is documented to walk the paths in lexical order, iterating
+	// in the reverse order ensures that files are always Lchowned before their parent directory
```

Comment:
Why is this necessary? Collecting a slice of unbounded size of paths doesn't feel great.

Reply:
It probably isn't 🤔 This was copied from the existing function that used to be in utils, but I don't see any reason to keep this behavior